File: Frontend/src/events/converters/parties/digimons/vital.converter.ts
import type { DeepRequired } from "@/events/dto/deep-required";
import type { VitalDTO } from "@/events/dto/parties/digimons/vital.dto";
import type { Vital } from "@/models";

export class VitalConverter {
  public static convert(vitalDto: DeepRequired<VitalDTO>): Vital {
    return {
      current: vitalDto.current,
      max: vitalDto.max,
    };
  }
}

File: Backend/Events/Converters/Parties/Digimons/VitalConverter.cs
using Backend.Domain.Models.Parties.Digimons;
using Backend.Events.DTO.Parties.Digimons;

namespace Backend.Events.Converters.Parties.Digimons;

public static class VitalConverter
{
    public static VitalDTO ToDTO(Vital vital) => new()
    {
        Current = vital.Current,
        Max = vital.Max
    };
}
